nixos/modules/services/networking/v2raya.nix

```
@@ -1,49 +1,117 @@
+          v2rayA.
+        '';
+      };
+
+      v2rayPath = mkOption {
+        type = with types; nullOr path;
+        default = null;
+        example = literalExpression ''"''${pkgs.xray}/bin/xray"'';
+        description = ''
+          v2ray executable file path. If set to `null`, will
+          use v2ray provided in v2rayA wrapper. It can be
+          modified to the file path of other v2ray branches
+          such as xray.
+        '';
+      };
+
+      logFile = mkOption {
+        type = types.path;
+        default = "/var/log/v2raya/v2raya.log";
+        description = ''
+          The path of v2rayA log file.
+        '';
+      };
+
+      environment = mkOption {
+        type = types.attrsOf lib.types.str;
+        default = { };
+        example = literalExpression ''
+          {
+            V2RAYA_PLUGINLISTENPORT = "32346";
+            V2RAYA_PASSCHECKROOT = "1";
+            V2RAYA_FORCE_IPV6_ON = "1";
+          }
+        '';
+        description = ''
+          Environment variables added to v2raya systemd service.
+        '';
+      };
     };
   };
 
-  config = mkIf config.services.v2raya.enable {
-    environment.systemPackages = [ pkgs.v2raya ];
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
 
-    systemd.services.v2raya =
-      let
-        nftablesEnabled = config.networking.nftables.enable;
-        iptablesServices = [
+    systemd.services.v2raya = {
+      unitConfig = {
+        Description = "v2rayA service";
+        Documentation = "https://github.com/v2rayA/v2rayA/wiki";
+        After = [
+          "network.target"
+          "network-online.target"
+          "nss-lookup.target"
           "iptables.service"
-        ] ++ optional config.networking.enableIPv6 "ip6tables.service";
-        tableServices = if nftablesEnabled then [ "nftables.service" ] else iptablesServices;
-      in
-      {
-        unitConfig = {
-          Description = "v2rayA service";
-          Documentation = "https://github.com/v2rayA/v2rayA/wiki";
-          After = [
-            "network.target"
-            "nss-lookup.target"
-          ] ++ tableServices;
-          Wants = [ "network.target" ];
-        };
-
-        serviceConfig = {
-          User = "root";
-          ExecStart = "${getExe pkgs.v2raya} --log-disable-timestamp";
-          Environment = [ "V2RAYA_LOG_FILE=/var/log/v2raya/v2raya.log" ];
-          LimitNPROC = 500;
-          LimitNOFILE = 1000000;
-          Restart = "on-failure";
-          Type = "simple";
-        };
-
-        wantedBy = [ "multi-user.target" ];
-        path = with pkgs; [ iptables bash iproute2 ]; # required by v2rayA TProxy functionality
+          "ip6tables.service"
+          "nftables.service"
+        ];
+        Wants = [ "network.target" ];
       };
+
+      serviceConfig = {
+        User = "root";
+        ExecStart = "${getExe cfg.package} --log-disable-timestamp";
```

Maybe we should add `--nftables-support` when `networking.nftables.enable` is enabled.
V2rayA doesn't seem to be able to auto-detect whether to use nftables on NixOS。
```suggestion
        ExecStart = "${getExe cfg.package} --log-disable-timestamp";
          + lib.optionalString config.networking.nftables.enable " --nftables-support";
```

---

However, I found that xray core does not support this option.
Maybe we should enable nftables through the environment variable `V2RAYA_NFTABLES_SUPPORT`?